pelican/tests/test_generators.py

```diff
@@ -698,21 +713,43 @@ def test_tag_and_category_links_on_generated_pages(self):
         Test to ensure links of the form {tag}tagname and {category}catname
         are generated correctly on pages
         """
-        settings = get_settings(filenames={})
+        settings = get_settings()
         settings['PAGE_PATHS'] = ['TestPages']    # relative to CUR_DIR
         settings['CACHE_PATH'] = self.temp_cache
         settings['DEFAULT_DATE'] = (1970, 1, 1)
+        context = get_context(settings)
 
         generator = PagesGenerator(
-            context=settings.copy(), settings=settings,
+            context=context, settings=settings,
             path=CUR_DIR, theme=settings['THEME'], output_path=None)
         generator.generate_context()
-        pages_by_title = {p.title: p.content for p in generator.pages}
```

Comment:
This line (pre-existing) in combination with the `{static}` and `{attach}` labels in `pelican/tests/TestPages/page_with_static_links.md` (introduced in this commit) caused `test_basic_generation_works` in `test_pelican.py` to fail on line 110, with a message that the log message is only found once instead of 3 times.

I spent hours on trying to find out how on earth this should be possible but didn't manage to narrow it down further than this. It looks like somehow the cache or the logs are polluted from this test, but I am absolutely befuddled how evaluating two extra static links here should reduce the number of log messages about three completely different static links in a test running on a  _different subtree of the project_. It's a complete mystery to me.